plugins/events-node/api-report.md

```diff
@@ -0,0 +1,106 @@
+## API Report File for "@backstage/plugin-events-node"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { ExtensionPoint } from '@backstage/backend-plugin-api';
+import { Request as Request_2 } from 'express';
+
+// @public
+export interface EventBroker {
+  publish(params: EventParams): Promise<void>;
+  subscribe(
+    ...subscribers: Array<EventSubscriber | Array<EventSubscriber>>
+  ): void;
+}
+
+// @public (undocumented)
+export interface EventParams {
+  eventPayload: unknown;
+  metadata?: Record<string, string | string[] | undefined>;
+  topic: string;
+}
+
+// @public
+export interface EventPublisher {
+  // (undocumented)
+  setEventBroker(eventBroker: EventBroker): Promise<void>;
+}
+
+// @public
+export abstract class EventRouter implements EventPublisher, EventSubscriber {
+  // (undocumented)
+  protected abstract determineDestinationTopic(
+    params: EventParams,
+  ): string | undefined;
+  // (undocumented)
+  onEvent(params: EventParams): Promise<void>;
+  // (undocumented)
+  setEventBroker(eventBroker: EventBroker): Promise<void>;
+  // (undocumented)
+  abstract supportsEventTopics(): string[];
+}
+
+// @alpha (undocumented)
+export interface EventsExtensionPoint {
+  // (undocumented)
+  addHttpPostIngress(options: HttpPostIngressOptions): void;
+  // (undocumented)
+  addPublishers(
+    ...publishers: Array<EventPublisher | Array<EventPublisher>>
+  ): void;
+  // (undocumented)
+  addSubscribers(
+    ...subscribers: Array<EventSubscriber | Array<EventSubscriber>>
+  ): void;
+  // (undocumented)
+  setEventBroker(eventBroker: EventBroker): void;
+}
+
+// @alpha (undocumented)
+export const eventsExtensionPoint: ExtensionPoint<EventsExtensionPoint>;
+
+// @public
+export interface EventSubscriber {
+  onEvent(params: EventParams): Promise<void>;
+  supportsEventTopics(): string[];
+}
+
+// @public (undocumented)
+export interface HttpPostIngressOptions {
+  // (undocumented)
+  topic: string;
+  // (undocumented)
+  validator?: RequestValidator;
+}
+
+// @public
+export interface RequestRejectionDetails {
+  // (undocumented)
+  payload: unknown;
+  // (undocumented)
+  status: number;
+}
+
+// @public
+export interface RequestValidationContext {
+  reject(details?: Partial<RequestRejectionDetails>): void;
+}
+
+// @public
+export type RequestValidator = (
+  request: Request_2,
```

Comment:
A thought here is, sometimes it's been problematic to expose the Request type since it turns out to be very subtly different and type-incompatible between the node version and node-fetch and cross-fetch and whatnot. Stuff that generally isn't of interest, like does it have `signal` or what's the actual shape of the Headers type in complex cases.

It COULD have been safest to have the http ingress class extract all of the interesting parts of the request first, into a fixed shape (body and headers only for starters perhaps?) and passing that into the validators. But I'm not sure if it's worth it. Just brain dumping a bit.

Reply:
PR #14691